web/authentication/exceptor_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authentication
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestPathAuthenticationExceptor_IsExcepted(t *testing.T) {
+	ts := []struct {
+		Name             string
+		ExcludedPaths    []string
+		URI              string
+		ExpectedExcepted bool
+	}{
+		{
+			Name:             "Path not excepted",
+			ExcludedPaths:    []string{"/somepath"},
+			URI:              "/someotherpath",
+			ExpectedExcepted: false,
+		},
+		{
+			Name:             "Exact path excepted (single)",
+			ExcludedPaths:    []string{"/somepath"},
+			URI:              "/somepath",
+			ExpectedExcepted: true,
+		},
```

Comment:
Imo `/somepath` shouldn't be considered a prefix of `/somepath/someotherpath` in either case here as it's not a subpath, but overall specifying whether subpaths should be excluded is a good point and I added a test for now expecting `/somepath` to not be excluded when `/` is excluded.

If you see a use case behind excluding prefixes I can add it, although imo it would have to only apply to subpaths and be very explicit to avoid potential errors from the user side. Perhaps explicitly specifying an `Exact` or `Prefix` kind, like [k8s ingress does](https://kubernetes.io/docs/concepts/services-networking/ingress/#path-types)? Let me know what you think, personally I'm fine with only allowing exact paths.

Reply:
No need to add support for excluding prefixes, I just wanted to make sure we are explicit about the semantics and have a test to enforce them – what I want to avoid is that we _accidentally_ turn this into a prefix-match at some point in the future. I see that there's a test for this now, so I'm happy 😄